.github/workflows/reusable-windows-ci.yml

```diff
@@ -0,0 +1,778 @@
+          'LibP11_ROOT_64=' + $libp11_path | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+
+          # Wait until installation is completed
+          Wait-Process -Name msiexec -ErrorAction SilentlyContinue
+          del $msiHSM
+
+          # Set up OpenSSL
+          $module_path = gci -Path $Env:SOFTHSM2_ROOT -Recurse -Include '*.dll' | ? FullName -match 64
+          $contents = gc $config;
+
+          $header = "# HSM test ancillary configuration",
+                    "openssl_conf = openssl_init"
+
+          $footer = "[openssl_init]",
+                    "engines = engine_section",
+                    "",
+                    "[engine_section]",
+                    "pkcs11 = pkcs11_section",
+                    "",
+                    "[pkcs11_section]",
+                    "engine_id = pkcs11",
+                    ("dynamic_path =" + ("$libp11_path/pkcs11.dll" -replace "\\","/")),
+                    ("MODULE_PATH =" + ($module_path.FullName -replace "\\","/")),
+                    "init = 0"
+
+          $header_line = ($contents | sls '^HOME')[0].LineNumber;
+          ($contents[0..$header_line] + $header +  $contents[$header_line..$contents.count] + $footer) |
+          % { $_.TrimStart() } | Out-File $config -Encoding OEM;
+
+          # lead openssl to the right config file
+          $Env:OPENSSL_CONF=$config
+
+          # introduce openssl binary in the path for the lower check
+          $Env:Path += ";$Env:OPENSSL64_ROOT\bin"
+
+          # check if is working
+          openssl engine pkcs11 -t
+
+      - name: Prepare colcon workspace
+        run: |
+          # Get some convenient tools
+          Install-Module -Name ConvertBase64Strings -Force -AllowClobber
+          Import-Module -name ConvertBase64Strings -Prefix CI
+          Install-Module powershell-yaml -Force
+
+          # Check if git is behaving
+          $checker = @{
+            Begin = {$errores = 0}
+            Process = {if($_ -is [Management.Automation.ErrorRecord] -and $_.Exception.Message -match "^(error|fatal):") {++$errores} }
+            End = { $errores -eq 0 }
+          }
+          $assert = {if($_ -is [Management.Automation.ErrorRecord] -and $_.Exception.Message -match "^(error|fatal):")
+                        { Write-Error $_ } else { Write-Output $_ } }
+
+          # Download eprosima-CI repo and checkout current branch
+          git clone https://github.com/eProsima/eprosima-CI.git eprosima-CI 2>&1 | % $assert
+          pushd eprosima-CI
+          # locally master is main
+          git symbolic-ref refs/heads/master refs/heads/main 2>&1 | % $assert
+
+          # selecting specific metas & depends
+          $repo_branch = "${{ inputs.force-meta-branch }}"
+
+          if($repo_branch)
+          {
+            # 1. Externally specified as nightlies
+            git checkout $repo_branch 2>&1 | % $assert
+          }
+          else
+          {
+            # 2. Use heuristics to choose
```

Comment:
Now that I'm double checking, all this decision making depends on a file structure on the eprosima-ci repository that is not there. Not even in main branch.

Reply:
I've changes the approach in cbbd902